src/adam_core/propagator/adam_assist.py

```diff
@@ -517,10 +523,127 @@ def _detect_impacts(
+            propagated_orbits_barycentric = propagated_orbits.set_column(
+                "coordinates",
+                transform_coordinates(
+                    propagated_orbits.coordinates,
+                    CartesianCoordinates,
+                    frame_out="ecliptic",
+                    origin_out=OriginCodes.SOLAR_SYSTEM_BARYCENTER,
+                ),
+            )
+            observers_barycentric = observers.set_column(
+                "coordinates",
+                transform_coordinates(
+                    observers.coordinates,
+                    CartesianCoordinates,
+                    frame_out="ecliptic",
+                    origin_out=OriginCodes.SOLAR_SYSTEM_BARYCENTER,
+                ),
+            )
+
+            num_orbits = len(propagated_orbits_barycentric.orbit_id.unique())
+            observer_coordinates = np.tile(
+                observers_barycentric.coordinates.values, (num_orbits, 1)
+            )
+            observer_codes = np.tile(observers.code.to_numpy(zero_copy_only=False), num_orbits)
+
+            propagated_orbits_aberrated, light_time = self._add_light_time(
+                propagated_orbits_barycentric, 
+                observers_barycentric, 
+                lt_tol=lt_tol, 
+            )
+
+            topocentric_coordinates = CartesianCoordinates.from_kwargs(
+                x=propagated_orbits_aberrated.coordinates.values[:,0] - observers_barycentric.coordinates.values[:,0],
+                y=propagated_orbits_aberrated.coordinates.values[:,1] - observers_barycentric.coordinates.values[:,1],
+                z=propagated_orbits_aberrated.coordinates.values[:,2] - observers_barycentric.coordinates.values[:,2],
+                vx=propagated_orbits_aberrated.coordinates.values[:,3] - observers_barycentric.coordinates.values[:,3],
+                vy=propagated_orbits_aberrated.coordinates.values[:,4] - observers_barycentric.coordinates.values[:,4],
+                vz=propagated_orbits_aberrated.coordinates.values[:,5] - observers_barycentric.coordinates.values[:,5],
+                covariance = None,
+                time=propagated_orbits_aberrated.coordinates.time,
+                origin=Origin.from_kwargs(code=observer_codes),
+                frame="ecliptic",
+            )
+
+            spherical_coordinates = SphericalCoordinates.from_cartesian(topocentric_coordinates)
+            light_time = np.array(light_time)
+
+            spherical_coordinates = transform_coordinates(
+                spherical_coordinates, SphericalCoordinates, frame_out="equatorial"
+            )
+
+            if isinstance(orbits, Orbits):
+
+                ephemeris = Ephemeris.from_kwargs(
+                    orbit_id=propagated_orbits_barycentric.orbit_id,
+                    object_id=propagated_orbits_barycentric.object_id,
+                    coordinates=spherical_coordinates,
+                    light_time=light_time,
```

Comment:
I notice we're not using the `aberrated_coordinates` column. I'm guessing it was decided that was too confusing?